test/conftest.py

```diff
@@ -28,6 +31,77 @@ def pytest_addoption(parser: pytest.Parser) -> None:
     )
 
 
+def docker_warmup(request: pytest.FixtureRequest) -> None:
+    machine = request.config.getoption("--run-emulation", default=None)
+    if machine is None:
+        archs = [arch.value for arch in Architecture.auto_archs("linux")]
+    elif machine == "all":
+        archs = EMULATED_ARCHS
+    else:
+        archs = [machine]
+    options = Options(
+        platform="linux",
+        command_line_arguments=CommandLineArguments.defaults(),
+        env={},
+        defaults=True,
+    )
+    build_options = options.build_options(None)
+    assert build_options.manylinux_images is not None
+    assert build_options.musllinux_images is not None
+    images = [build_options.manylinux_images[arch] for arch in archs] + [
+        build_options.musllinux_images[arch] for arch in archs
+    ]
+    for image in images:
+        try:
+            command = (
+                "manylinux-interpreters ensure $(manylinux-interpreters list 2>/dev/null | grep -v graalpy) &&"
+                "cpython3.13 -m pip download -d /tmp setuptools wheel pytest"
+            )
+            container_id = subprocess.run(
+                ["docker", "create", image, "bash", "-c", command],
+                text=True,
+                check=True,
+                stdout=subprocess.PIPE,
+            ).stdout.strip()
+            subprocess.run(["docker", "start", container_id], check=True, stdout=subprocess.DEVNULL)
+            exit_code = subprocess.run(
+                ["docker", "wait", container_id], text=True, check=True, stdout=subprocess.PIPE
+            ).stdout.strip()
+            if exit_code == "0":
+                subprocess.run(
+                    ["docker", "commit", container_id, image], check=True, stdout=subprocess.DEVNULL
+                )
+            subprocess.run(["docker", "rm", container_id], check=True, stdout=subprocess.DEVNULL)
```

Comment:
Might be polite to run this in a `finally:` block

Reply:
done in the commit that enforce the warm-up does not fail